frontend/src/app/services/eta.service.ts

```diff
@@ -143,7 +145,28 @@ export class EtaService {
 
     if (!isAccelerated) {
       const blocks = mempoolPosition.block + 1;
-      const wait = da.adjustedTimeAvg * (mempoolPosition.block + 1);
+
+      // Estimate future incoming tx rate from mempool statistics
+      let vsizePerSecond = Math.min(
+        this.estimateVsizePerSecond(tx, mempoolStats),
+        0.95 * this.stateService.blockVSize / da.adjustedTimeAvg * 1000
+      );
+
+      // Count the number of blocks until we expect this tx to be mined
+      let blocksUntilMined = 0;
+      let mined = false;
+      let vsize = mempoolPosition.vsize + this.stateService.blockVSize * mempoolPosition.block;
+      // This loop will always terminate because we cap vsizePerSecond to 0.95 * maxCapacity
+      while (!mined) {
+        vsize = vsize + vsizePerSecond * da.adjustedTimeAvg / 1000 - this.stateService.blockVSize;
+        if (vsize + tx.weight / 8 < 0) { // Means that our tx fits in expected next block
```

Comment:
why `tx.weight / 8`?

Reply:
It seems `mempoolPosition.vsize` is the sum of the vsizes of the transactions ahead, plus half of the transaction vsize itself. 
So to check that the transaction fits in a block with `mempoolPosition.vsize` ahead, we check that `mempoolPosition.vsize` plus the other half of the transaction vsize `(tx.weight / 4) / 2` is lower than `blockVsize`... it's probably nitpicking since this value is at most 50 kvB in our mempool 😅

Reply:
ah ok, that makes sense